packages/orm/src/AliasAssigner.ts

```diff
@@ -8,13 +8,19 @@ export class AliasAssigner {
     this.getAlias = this.getAlias.bind(this);
     // If we're assigning aliases into an existing query, get the current assignments
     if (query) {
-      for (const table of query.tables) {
-        // Ignore CTI base/sub aliases
-        if (table.alias.includes("_")) continue;
-        const abbrev = abbreviation(table.table);
-        const i = this.#aliases[abbrev] || 0;
-        const j = Number(table.alias.replace(abbrev, "")) + 1;
-        this.#aliases[abbrev] = Math.max(i, j);
+      const todo = [query];
```

Comment:
Rewrote this to crawl through nested `ParsedFindQuery`s if necessary (nested `ParsedFindQuery`s is how lateral joins are implemented now).